src/Bicep.RegistryModuleTool/ModuleValidators/BicepMetadataValidator.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.RegistryModuleTool.Exceptions;
+using Bicep.RegistryModuleTool.Extensions;
+using Bicep.RegistryModuleTool.ModuleFiles;
+using Microsoft.Extensions.Logging;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Bicep.RegistryModuleTool.ModuleValidators
+{
+    public sealed class BicepMetadataValidator : IModuleFileValidator
+    {
+        private readonly ILogger logger;
+
+        private readonly MainBicepFile latestMainBicepFile;
+
+        public BicepMetadataValidator(ILogger logger, MainBicepFile latestMainBicepFile)
+        {
+            this.logger = logger;
+            this.latestMainBicepFile = latestMainBicepFile;
+        }
+
+        public void Validate(MainArmTemplateFile templateFile)
+        {
+            this.logger.LogInformation("Verifying Bicep metadata...");
+            var errorMessageBuilder = new StringBuilder();
+
+            var moduleName = templateFile.NameMetadata;
+            var moduleDescription = templateFile.DescriptionMetadata;
+            var moduleOwner = templateFile.OwnerMetadata;
+
+            var missingMetadata = new List<string>();
+            if (moduleName is null)
+            {
+                missingMetadata.Add(MainBicepFile.ModuleNameMetadataName);
+            }
+            if (moduleDescription is null)
+            {
+                missingMetadata.Add(MainBicepFile.ModuleDescriptionMetadataName);
+            }
+            if (moduleOwner is null)
+            {
+                missingMetadata.Add(MainBicepFile.ModuleOwnerMetadataName);
+            }
+            if (missingMetadata.Count > 0)
+            {
+                errorMessageBuilder.AppendLine($"The file \"{latestMainBicepFile.Path}\" is invalid. The following metadata statements are missing:");
+                foreach (var entry in missingMetadata)
+                {
+                    errorMessageBuilder.AppendLine($"  - metadata {entry}");
+                }
+
+                throw new InvalidModuleException(errorMessageBuilder.ToString().Trim());
+            }
+
+            ValidateMinLength("name", moduleName!, 10);
+            ValidateMaxLength("name", moduleName!, 60);
+
+            ValidateMinLength("description", moduleDescription!, 10);
+            ValidateMaxLength("description", moduleDescription!, 120);
+
+            ValidateAgainstRegex(
+                "owner",
+                moduleOwner!,
+                new Regex("^(?:Azure\\/)?[a-zA-Z\\d](?:[a-zA-Z\\d]|-(?=[a-zA-Z\\d])){0,38}$"),
```

Comment:
We should consider using a regex source generator (https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-source-generators).

Reply:
Interesting.  137e168f745a9968e4d86b5dc57a67cfda07dadb